src/test/java/com/revature/servicestests/SearchServiceTest.java

```diff
@@ -62,130 +66,78 @@
-		String format = first.getFormat();
+		//Valid testing.
+		Set<Content> filtered = ss.filter(title, format, mlist);
+		boolean validFilter = filtered.contains(content);
 		
-		int rows = JdbcTestUtils.countRowsInTableWhere(template, "content", String.format("format = '%s'", format));
-		assertEquals(rows, ss.filterContentByFormat(format).size());
-		assertEquals(0, ss.filterContentByFormat("NON EXISTENT FORMAT").size());
-	}	
-
-	@Test
-	@Commit
-	@Order(4)
-	void testFilterContentBySubjects() {
-		Set<Module> lst = ms.getAllModules();
-		Iterator<Module> iter = lst.iterator();		
+		filtered = ss.filter(badtitle, format, mlist);
+		boolean badTitleFilter = filtered.contains(content);
 		
-		int flamingModId = -1;
-		int elmoModId = -1;
-		while(iter.hasNext()) {			
-			Module current = iter.next();
-			if(current.getSubject().equals("FIRST TEST MODULE")) {
-				flamingModId = current.getId();
-			}
-			if(current.getSubject().equals("SECOND TEST MODULE")) {
-				elmoModId = current.getId();
-			}				
-		}
+		filtered = ss.filter(title, badformat, mlist);
+		boolean badFormatFilter = filtered.contains(content);
 		
-		List<Integer> modIdArray = new ArrayList<Integer>();
+		mlist.clear();
+		mlist.add(module3.getId());
+		filtered = ss.filter(title, format, mlist);
+		boolean badModuleFilter = filtered.contains(content);
 		
-		// to test filter content with only 1 subject
-		modIdArray.add(flamingModId);
-		int rows = JdbcTestUtils.countRowsInTableWhere(template, "link", String.format("fk_m = '%d'", flamingModId));
-		assertEquals(rows, ss.filterContentBySubjects(modIdArray).size());
 		
-		// to test filter content with more than one subject
-		modIdArray.add(elmoModId);
-		assertEquals(1, ss.filterContentBySubjects(modIdArray).size());
+		//Cleanup courtesy of @Transactional and @Rollback.
+		assertTrue(titleTest);
+		assertTrue(formatTest);
+		assertTrue(subjectTestContains);
+		assertTrue(moduleIdTestContains);
 		
-		// tests filter content by Subject with non-existent subject
-		List<Integer> modIdArray2 = new ArrayList<Integer>();
-		modIdArray2.add(-1);
-		assertEquals(0, ss.filterContentBySubjects(modIdArray2).size());
-	}
-
-	@Test
-	@Commit
-	@Order(5)
-	void testGetContentByModuleId() {
-		Set<Module> allModules= ms.getAllModules();
-		Iterator<Module> iter = allModules.iterator();
-		Module first = iter.next();
-		int MID = first.getId();
-				
-		assertThrows(NoSuchElementException.class, ()->{ss.getContentByModuleId(-1);});
-		assertNotNull(ss.getContentByModuleId(MID));
-	}
-	
-	@Test
-	@Commit
-	@Order(6)
-	void deleteTestData() {
-		JdbcTestUtils.deleteFromTableWhere(template, "module", "subject = 'FIRST TEST MODULE'");
-		JdbcTestUtils.deleteFromTableWhere(template, "module", "subject = 'SECOND TEST MODULE'");
-		JdbcTestUtils.deleteFromTableWhere(template, "content", "title = 'FIRST TEST CONTENT'");
+		assertTrue(validFilter);
+		assertFalse(badTitleFilter);
+		assertFalse(badFormatFilter);
+		assertFalse(badModuleFilter);
```

Comment:
Each test should test one thing at a time, this appears to test 8.

Reply:
Tests have been split.